golem-service-base/src/service/plugin_wasm_files.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2024-2025 Golem Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{path::PathBuf, sync::Arc};
+
+use crate::replayable_stream::{ContentHash, ReplayableStream};
+use crate::storage::blob::{BlobStorage, BlobStorageNamespace};
+use bytes::Bytes;
+use golem_common::model::plugin::PluginWasmFileKey;
+use golem_common::model::AccountId;
+use tracing::debug;
+
+const LIBRARY_PLUGIN_STORE_LABEL: &str = "library_plugins";
+
+/// Service for storing plugin wasm files.
+pub struct PluginWasmFilesService {
```

Comment:
In principle this could be separated for library and app plugins.

Alternatively, as the this is using content hashing, we could also merge it with initial component files and store everything here, but that might make debugging harder.

Want me to go for either of those options?

Reply:
I think it's ok like this